src/store/modules/events/index.ts

```diff
@@ -538,21 +580,70 @@ const module: Module<EventsModuleState, RootState> = {
         return;
       }
 
-      const repetition = event.repetition;
+      /**
+       * Default value for event repetition, in case of new delta format, we will patch it
+       */
+      let repetition = event.repetition;
+
+      /**
+       * Check if delta format is new
+       */
+      const isNewDeltaFormat = !event.repetition.payload;
+
+      /**
+       * If delta format is new, apply delta to the event payload
+       */
+      if (event.repetition.delta && isNewDeltaFormat) {
+        const eventPayload = cloneDeep(event.payload);
+
+        repetition = {
+          ...event.repetition,
+          payload: patch({
+            left: eventPayload,
+            delta: JSON.parse(event.repetition.delta),
+          }),
+        };
+      } else if (isNewDeltaFormat) {
+        /**
+         * If delta format is new, set the event payload to the original event payload in case if delta is null
+         */
+        repetition = {
+          ...event.repetition,
+          payload: event.payload,
+        };
+      }
+
+      /**
+       * Update event repetition
+       */
+      event.repetition = repetition;
 
       filterBeautifiedAddons([ event ]);
       filterBeautifiedAddons([ event.repetition ]);
 
       /**
-       * Updates or sets event's fetched payload in the state
+       * Save to the state, if delta format is new, otherwise assemble event payload and save to the state
        */
-      commit(MutationTypes.UpdateEvent, {
-        projectId,
-        event: {
-          ...event,
-          payload: repetition ? repetitionAssembler(event.payload, repetition.payload) as HawkEventPayload : event.payload,
-        },
-      });
+      if (isNewDeltaFormat) {
+        commit(MutationTypes.UpdateEvent, {
+          projectId,
+          event: {
+            ...event,
+            payload: repetition.payload,
+          },
+        });
+      } else {
+        /**
+         * Save to the state, if delta format is old, assemble event payload and save to the state
+         */
+        commit(MutationTypes.UpdateEvent, {
+          projectId,
+          event: {
+            ...event,
+            payload: repetition ? repetitionAssembler(event.payload, repetition.payload) as HawkEventPayload : event.payload,
```

Comment:
it is expected to see `processRepetitionPayload` since now it is responsible for composing a payload

Reply:
done